### PR TITLE
Refactor/benchmarking

### PR DESCRIPTION
**Pull request by Eikix**
Time spent on this PR:

Resolves: #694 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

# Does this introduce a breaking change?

- [ ] Yes
- [ ] No

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot-rpc/697)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,7 +17,7 @@ deploy-kakarot:
 	cd lib/kakarot && STARKNET_NETWORK=$(STARKNET_NETWORK) poetry run python ./scripts/deploy_kakarot.py && cd ..
 
 run-dev:
-	PROXY_ACCOUNT_CLASS_HASH=$(shell jq -r '.proxy' ./lib/kakarot/deployments/$(STARKNET_NETWORK)/declarations.json) CONTRACT_ACCOUNT_CLASS_HASH=$(shell jq -r '.contract_account' ./lib/kakarot/deployments/$(STARKNET_NETWORK)/declarations.json) EXTERNALLY_OWNED_ACCOUNT_CLASS_HASH=$(shell jq -r '.externally_owned_account' ./lib/kakarot/deployments/$(STARKNET_NETWORK)/declarations.json) KAKAROT_ADDRESS=$(shell jq -r '.kakarot.address' ./lib/kakarot/deployments/$(STARKNET_NETWORK)/deployments.json) RUST_LOG=trace cargo run
+	PROXY_ACCOUNT_CLASS_HASH=$(shell jq -r '.proxy' ./lib/kakarot/deployments/$(STARKNET_NETWORK)/declarations.json) CONTRACT_ACCOUNT_CLASS_HASH=$(shell jq -r '.contract_account' ./lib/kakarot/deployments/$(STARKNET_NETWORK)/declarations.json) EXTERNALLY_OWNED_ACCOUNT_CLASS_HASH=$(shell jq -r '.externally_owned_account' ./lib/kakarot/deployments/$(STARKNET_NETWORK)/declarations.json) KAKAROT_ADDRESS=$(shell jq -r '.kakarot.address' ./lib/kakarot/deployments/$(STARKNET_NETWORK)/deployments.json) RUST_LOG=trace cargo run --bin kakarot-rpc
```
Comment (greged93) [inline Makefile:20]: I don't understand why this diff is present, as main should have it already

File: Makefile [hunk 2]
```diff
@@ -54,7 +54,13 @@ test: dump-katana
 test-coverage:
 	cargo llvm-cov nextest --all-features --workspace --lcov --output-path lcov.info
 
-test-examples:
-	hurl $(HURL_FILES)
+# Make sure to have a Kakarot RPC running and the correct port set in your .env and an underlying Starknet client running.
+# Use benchmark-madara if the underlying Starknet node is Madara
```
Comment (greged93) [inline Makefile:58]: You can remove this line, should be clear enough

File: Makefile [hunk 3]
```diff
@@ -54,7 +54,13 @@ test: dump-katana
...
+benchmark-madara:
+	cd benchmarks && bun i && bun run benchmark-madara
+# Use benchmark-katana if the underlying Starknet node is Katana
```
Comment (greged93) [inline Makefile:61]: Same

File: benchmarks/index.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  try {
+    await wallet.sendTransaction({
+      to: recipient,
+      nonce,
+      value: SEND_AMOUNT,
+    });
+  } catch (e) {
+    // We expect to get an error:
+    // @TODO: the returned hash did not match
+    // <https://github.com/ethers-io/ethers.js/issues/4233>
```
Comment (greged93) [inline benchmarks/index.ts:75]: Maybe just match on the error? To be sure we are receiving a invalid transaction hash error.

File: Makefile (hunk 3 above)
Comment (Eikix) [reply Makefile]: Ok!

File: Makefile (hunk 1 above)
Comment (Eikix) [reply Makefile:20]: idk

File: benchmarks/index.ts (hunk 4 above)
Comment (Eikix) [reply benchmarks/index.ts:73]: Ok!

**Review by greged93 — APPROVED** [review]
lgtm

**Pull request merged**